### DISCUSSION TITLE
add-more-professional-demo-presentation-capabilities-to-vitepress

### DISCUSSION THREAD

**Discussion by bowencool** [bowencool/blog#44] [Announcements]
# add-more-professional-demo-presentation-capabilities-to-vitepress

为 vitepress 添加更专业的 Demo 演示能力

https://blog.bowen.cool/posts/add-more-professional-demo-presentation-capabilities-to-vitepress

**Comment by MrWeilian** [comment]
666 学习了 组件库的下一步就是搞demo简化了

**Comment by stevending1st** [comment]
vitepress 支持 vite 插件，在 `vitepress <命令> <文件夹>` 命令中 `<文件夹>`  的根目录创建 `vite.config.js` 即可。
例如：
``` text
.
├─ docs
│  ├─ .vitepress
│  │  ├─ theme
│  │  │  └─ index.js
│  │  └─ config.js
│  ├─ vite.config.js   <------ 创建
│  └─ index.md
└─ package.json
```
更多例子可参考：

https://github.com/vueuse/vueuse/blob/4dea4be6f46a85a880ccd08954c84febf332187a/packages/vite.config.ts#L33-L35

**Comment by bowencool** [comment]
> vitepress 支持 vite 插件，在 `vitepress <命令> <文件夹>` 命令中 `<文件夹>` 的根目录创建 `vite.config.js` 即可。 例如：
> 
> ```
> .
> ├─ docs
> │  ├─ .vitepress
> │  │  ├─ theme
> │  │  │  └─ index.js
> │  │  └─ config.js
> │  ├─ vite.config.js   <------ 创建
> │  └─ index.md
> └─ package.json
> ```
> 
> 更多例子可参考：
> 
> https://github.com/vueuse/vueuse/blob/4dea4be6f46a85a880ccd08954c84febf332187a/packages/vite.config.ts#L33-L35

是我表达有点问题，我的意思是 vitepress 核心功能不支持插件扩展

**Comment by Zhaojing-0221** [comment]
必须在项目初始化的就使用这个插件吗？我现在已经有一个做的差不多的文档里，可以用yarn下载插件再使用吗？

**Comment by bowencool** [comment]
> 必须在项目初始化的就使用这个插件吗？我现在已经有一个做的差不多的文档里，可以用yarn下载插件再使用吗？

拷贝关键代码就行。vitepress 核心功能不支持插件扩展，所以暂时没有好的办法。
